hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java

```diff
@@ -268,6 +268,82 @@ public void trunc() {
 				.setInvariantType(doubleType)
 				.setArgumentListSignature( "(NUMERIC number[, INTEGER places])" )
 				.register();
+		functionRegistry.registerAlternateKey( "truncate", "trunc" );
+	}
+
+	/**
+	 * MySQL
+	 */
+	public void trunc_truncate() {
+		functionRegistry.registerUnaryBinaryPattern(
+				"trunc",
+				doubleType,
+				"truncate(?1,0)",
+				"truncate(?1,?2)",
+				NUMERIC, INTEGER,
+				typeConfiguration
+		).setArgumentListSignature( "(NUMERIC number[, INTEGER places])" );
+		functionRegistry.registerAlternateKey( "truncate", "trunc" );
+	}
+
+	/**
+	 * SQL Server
+	 */
+	public void trunc_round() {
+		functionRegistry.registerUnaryBinaryPattern(
+				"trunc",
+				doubleType,
+				"round(?1,0,1)",
+				"round(?1,?2,1)",
+				NUMERIC, INTEGER,
+				typeConfiguration
+		).setArgumentListSignature( "(NUMERIC number[, INTEGER places])" );
+		functionRegistry.registerAlternateKey( "truncate", "trunc" );
+	}
+
+	/**
+	 * Sybase
+	 */
+	public void trunc_floorPower() {
+		functionRegistry.registerUnaryBinaryPattern(
+				"trunc",
+				doubleType,
+				"sign(?1)*floor(abs(?1))",
+				"sign(?1)*floor(abs(?1)*power(10,?2))/power(10,?2)",
+				NUMERIC, INTEGER,
+				typeConfiguration
+		).setArgumentListSignature( "(NUMERIC number[, INTEGER places])" );
+		functionRegistry.registerAlternateKey( "truncate", "trunc" );
+	}
+
+	/**
+	 * PostgreSQL (only works if the second arg is constant, as it almost always is)
```

Comment:
PostgreSQL has a two arg `trunc` method. Am I missing the reason why you added this?
See https://www.postgresql.org/docs/8.1/functions-math.html

Reply:
The two-arg form only accepts `numeric`. It's right there in that link you posted.

Reply:
Did you try it? The docs are sometimes misleading with the types they name.

Reply:
> ERROR:  function round(double precision, integer) does not exist

Reply:
Hm you are right, but should we maybe render something different depending on the type? Seems "wasteful" to me to do all that `floor` and division stuff also for `numeric` typed expressions especially since round/trunc will mostly be done just on `numeric` typed columns anyway.

Reply:
I'm not motivated to spend time on that.

Reply:
> I'm not motivated to spend time on that.

(Since I care more about scientific computing than about financial computing, I care more about `double` than I do about `numeric` haha.)